rubix/core/rotation.py

```diff
@@ -43,29 +43,57 @@ def get_galaxy_rotation(config: dict):
         beta = config["galaxy"]["rotation"]["beta"]
         gamma = config["galaxy"]["rotation"]["gamma"]
 
-    def rotate_galaxy(inputs: dict[str, jax.Array], type: str = "face-on"):
+    def rotate_galaxy(rubixdata: object, type: str = "face-on"):
         logger.info(f"Rotating galaxy with alpha={alpha}, beta={beta}, gamma={gamma}")
 
-        # Get the inputs
-        coords = inputs["coords"]
-        velocities = inputs["velocities"]
-        masses = inputs["mass"]
-        halfmass_radius = inputs["halfmassrad_stars"]
-
-        # Rotate the galaxy
-        coords, velocities = rotate_galaxy_core(
-            positions=coords,
-            velocities=velocities,
-            masses=masses,
-            halfmass_radius=halfmass_radius,
-            alpha=alpha,
-            beta=beta,
-            gamma=gamma,
-        )
-
-        # Update the inputs
-        inputs["coords"] = coords
-        inputs["velocities"] = velocities
-        return inputs
+        if "stars" in config["data"]["args"]["particle_type"]:
```

Comment:
is this old code?

I feel like we had a imprvoved version on this where we refactored this into different funciont, since this is just copy past of almost the same code for both stars and gas.

we should refactor this

Reply:
Yes, this is old code, the most up to date version is currently on the branch final load gas particles